From 4767f24ea4c1e3822ce71a636c341e8ad8d07aa6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Wed, 15 Jun 2016 18:04:43 +0200
Subject: [PATCH] Limit number of included scripts in get_scripts.js.php
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This avoids potential DOS, the limit is same as we use for generating
the URLs.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 js/get_scripts.js.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/get_scripts.js.php b/js/get_scripts.js.php
index 8e828c91cc9..06705c5d4ea 100644
--- a/js/get_scripts.js.php
+++ b/js/get_scripts.js.php
@@ -36,7 +36,8 @@ function () {
 
 $_GET['scripts'] = json_decode($_GET['scripts']);
 if (! empty($_GET['scripts']) && is_array($_GET['scripts'])) {
-    foreach ($_GET['scripts'] as $script) {
+    // Only up to 10 scripts as this is what we generate
+    foreach (array_slice($_GET['scripts'], 0, 10) as $script) {
         // Sanitise filename
         $script_name = 'js';
 
